mono/mini/method-to-ir.c

```diff
@@ -2173,6 +2179,23 @@ mono_emit_call_args (MonoCompile *cfg, MonoMethodSignature *sig,
 	if (cfg->llvm_only)
 		tail = FALSE;
 
+#if defined (DEBUG_TAILCALL_BREAK_COMPILE) || defined (DEBUG_TAILCALL_BREAK_RUN) || defined (DEBUG_TAILCALL)
+	if (tail && mono_is_usermode_native_debugger_present ()) {
+
```

Comment:
Putting ifs inside #ifdefs confuses the indentation of some editors, so its discouraged.

Reply:
I'd have preferred the old #if 0 code, its just there for debugging, I don't think commented out code like that needs to be named with specific ifdefs.

Reply:
Ludovic did not like if 0, wanted a name, and I don't think they deserve the same name, really.

Reply:
In the end, it's up to you @vargaz.

Reply:
if and ifdef seem like pretty basic features that we should be able to use, together. I could make a const static gboolean and just use if, not idef. The compiler will see through it and optimize and the code will not be allowed to rot.

Reply:
Doing 
#ifdef 
  if ()
     ..;
#endif 
is ok, but doing
#  ifdef
   if ()
#endif 
is not.
The static gboolean approach is ok too.